app/main.py

```diff
@@ -1 +1,53 @@
-#  write your code here
+from typing import List
+
+
+class Animal:
+    def __init__(
+        self,
+        name: str,
+        appetite: int,
+        is_hungry: bool = True,
+    ) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        if self.is_hungry:
+            print(f"Eating {self.appetite} food points...")
+            self.is_hungry = False
+            return self.appetite
+        return 0
+
+    def __repr__(self) -> str:
+        return (
+            f"{self.__class__.__name__}("
+            f"{self.name}, appetite={self.appetite}, "
+            f"hungry={self.is_hungry})"
+        )
+
+
+class Cat(Animal):
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, appetite=3, is_hungry=is_hungry)
+
+    def catch_mouse(self) -> None:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, appetite=7, is_hungry=is_hungry)
+
+    def bring_slippers(self) -> None:
+        print("The slippers delivered!")
+
+
+def feed_animals(animals: List[Animal]) -> int:
+    total_food = 0
+    for animal in animals:
+        total_food += animal.feed()
+    return total_food
```

Comment:
This implementation works, but it violates a guideline from the project's checklist. According to checklist item https://github.com/mate-academy/py-feed-animals/pull/2 under 'Make Code Easier': "Don't use a `for` loop if you can use generator expression."

You should use a generator expression with the `sum()` function instead of a `for` loop to calculate the total. This approach is more concise and Pythonic.

Could you refactor this to use `sum()` and a generator expression?